mono/utils/memfuncs.c

```diff
@@ -221,3 +234,57 @@ mono_gc_memmove_atomic (void *dest, const void *src, size_t size)
 	else
 		mono_gc_memmove_aligned (dest, src, size);
 }
+
+guint64
+mono_determine_physical_ram_size (void)
+{
+#if defined (TARGET_WIN32)
+	MEMORYSTATUSEX memstat;
+
+	memstat.dwLength = sizeof (memstat);
+	GlobalMemoryStatusEx (&memstat);
+	return (guint64)memstat.ullTotalPhys;
+#elif defined (__NetBSD__) || defined (__APPLE__)
+#ifdef __NetBSD__
```

Comment:
Is this all from documentation? Or CoreCLR? Or moved from elsewhere? I guess moved.